tests/ImageSharp.Tests/Processing/Transforms/ProjectiveTransformTests.cs

```diff
@@ -0,0 +1,139 @@
+            nameof(KnownResamplers.NearestNeighbor),
+            nameof(KnownResamplers.Robidoux),
+            nameof(KnownResamplers.RobidouxSharp),
+            nameof(KnownResamplers.Spline),
+            nameof(KnownResamplers.Triangle),
+            nameof(KnownResamplers.Welch),
+        };
+
+        public static readonly TheoryData<TaperSide, TaperCorner> TaperMatrixData =
+            new TheoryData<TaperSide, TaperCorner>
+                {
+                    { TaperSide.Bottom, TaperCorner.Both },
+                    { TaperSide.Bottom, TaperCorner.LeftOrTop },
+                    { TaperSide.Bottom, TaperCorner.RightOrBottom },
+
+                    { TaperSide.Top, TaperCorner.Both },
+                    { TaperSide.Top, TaperCorner.LeftOrTop },
+                    { TaperSide.Top, TaperCorner.RightOrBottom },
+
+                    { TaperSide.Left, TaperCorner.Both },
+                    { TaperSide.Left, TaperCorner.LeftOrTop },
+                    { TaperSide.Left, TaperCorner.RightOrBottom },
+
+                    { TaperSide.Right, TaperCorner.Both },
+                    { TaperSide.Right, TaperCorner.LeftOrTop },
+                    { TaperSide.Right, TaperCorner.RightOrBottom },
+
+                };
+
+        public ProjectiveTransformTests(ITestOutputHelper output)
+        {
+            this.Output = output;
+        }
+
+        [Theory]
+        [WithTestPatternImages(nameof(ResamplerNames), 150, 150, PixelTypes.Rgba32)]
+        public void Transform_WithSampler<TPixel>(TestImageProvider<TPixel> provider, string resamplerName)
+            where TPixel : struct, IPixel<TPixel>
+        {
+            IResampler sampler = GetResampler(resamplerName);
+            using (Image<TPixel> image = provider.GetImage())
+            {
+                Matrix4x4 m = ProjectiveTransformHelper.CreateTaperMatrix(image.Size(), TaperSide.Right, TaperCorner.Both, .5F);
+
+                image.Mutate(i => { i.Transform(m, sampler); });
+
+                image.DebugSave(provider, resamplerName);
+                image.CompareToReferenceOutput(ValidatorComparer, provider, resamplerName);
+            }
+        }
+
+        [Theory]
+        [WithSolidFilledImages(nameof(TaperMatrixData), 30, 30, nameof(Rgba32.Red), PixelTypes.Rgba32)]
+        public void Transform_WithTaperMatrix<TPixel>(TestImageProvider<TPixel> provider, TaperSide taperSide, TaperCorner taperCorner)
+            where TPixel : struct, IPixel<TPixel>
+        {
+            var taperMatrixComparer = ImageComparer.TolerantPercentage(0.2f);
+            using (Image<TPixel> image = provider.GetImage())
+            {
+                Matrix4x4 m = ProjectiveTransformHelper.CreateTaperMatrix(image.Size(), taperSide, taperCorner, .5F);
+                image.Mutate(i => { i.Transform(m); });
+
+                string testOutputDetails = $"{taperSide}-{taperCorner}";
+                image.DebugSave(provider, testOutputDetails);
```

Comment:
@JimBobSquarePants added some tests to cover CreateTaperMatrix() parameters. I may get it wrong but the `XxxOrYyy` stuff might be the inverse of the intended. Can you check the output?

Reply:
I'll have to do some tests locally to ensure the output is expected. The class is based on a Skia example and and old Charles Petzold blog post.